src/lib/redux/siteData.ts

```diff
@@ -34,4 +35,8 @@ export const updateTariffPlanInRedux = (tariffPlan: Partial<TariffPlanRecord>) =
 
 export const updateSiteInRedux = (siteUpdates : Partial<SiteRecord>) => {
   store.dispatch(updateSite(siteUpdates))
-}
\ No newline at end of file
+}
+
+export const selectCurrentSiteProfit = () => {
+  return selectTotalAmountCollected(store.getState()) - selectAmountPurchaseRequestedApproved(store.getState());
```

Comment:
Selectors can build off other selectors via the `createSelector` function. You can look at `inventoryData` or `customerData` for inspiration, but should look something like

```
export const selectCurrentSiteProfit = createSelector(
selectTotalAmountCollected,
selectAmountPurchaseRequestedApproved,
(totalAmountCollected, totalAmountSpent) => totalAmountCollected - totalAmountSpent
```

Reply:
![image](https://user-images.githubusercontent.com/20329981/115972038-f8f01a00-a500-11eb-91f4-1d8ad97e2d1c.png)
tried that but it's giving me this error

Reply:
okok fair! Could you write a comment saying that `createSelector` isn't being in this case used because the needed selectors aren't initialized yet when the file is read?